projects/cdk/observables/pressed-observable.ts

```diff
@@ -3,15 +3,22 @@ import {filter, mapTo, startWith, switchMapTo, take} from 'rxjs/operators';
 import {mouseDragFinishFrom} from './mouse-drag-finish-from';
 import {typedFromEvent} from './typed-from-event';
 
-export function pressedObservable(element: Element): Observable<boolean> {
+export interface TuiPressedObservableOptions {
+    onlyTrusted: boolean;
+}
+
+export function pressedObservable(
+    element: Element,
+    {onlyTrusted}: TuiPressedObservableOptions = {onlyTrusted: true},
+): Observable<boolean> {
     const {ownerDocument} = element;
 
     if (!ownerDocument) {
         throw new Error('element does not have ownerDocument');
     }
 
     return typedFromEvent(element, 'mousedown').pipe(
-        filter(({isTrusted}) => isTrusted),
+        filter((event: MouseEvent) => event.isTrusted || !onlyTrusted),
```

Comment:
Why no destructuring?

Reply:
So, I thought a lot this weekend about how to reduce cognitive dive.

Reading this code `!onlyTrusted || isTrusted`:

it seems like here "* trusted" refers to different things, apparently "onlyTrusted" is about some mode, and "isTrusted" is about the current event. And it would probably be great to somehow indicate this in the name, so that it would be easier to navigate in similar names, but defining different behavior.

Reply:
@waterplea updated